Userland/Applications/Browser/IconBag.cpp

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2022, Dylan Katz <dykatz@uw.edu>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/String.h>
+#include <Applications/Browser/IconBag.h>
+
+namespace Browser {
+ErrorOr<IconBag*> IconBag::try_create()
```

Comment:
Plain pointers? I would suggest just doing this by-value. Since `TRY(try_create())` is a temporary, the compiler should implicitly generate a move constructor and use that to simply move all the `RefPtr`s anyway. No need to add more allocations when what we're trying to do is to keep allocations in check.